### DISCUSSION TITLE
Missing "Documentation" link for some packages that have documentation

### DISCUSSION THREAD

**Discussion by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#1864] [General]
I just did a full run-through of all package pages that have `doc_archives` in the database to check for obvious errors, and found two packages without a "Documentation" link at all!

* [Nifty-Markdown-Formatter](https://swiftpackageindex.com/NiftyTreeStudios/Nifty-Markdown-Formatter) (3CA0A035-46A1-4446-BC1E-D8B2697B7FC4)
* [swift-mujoco](https://swiftpackageindex.com/liuliu/swift-mujoco) (B38CD792-EAB2-40A3-9CC6-E173B5109842)

**Comment by daveverwer** [comment]
Interesting. I just investigated a little and both packages have empty JSON in their `doc_archives` field, which explains it!

It looks like [swift-mujoco](https://swiftpackageindex.com/liuliu/swift-mujoco) has a problem with their macOS build right now, and that's the platform being used for compilation. It looks to be something they are [trying to fix](https://github.com/liuliu/swift-mujoco/commit/2adff303e4b426f9d94832fe2c7ea65b1ea6c73c), so it should all fix up when that resolves.

**Comment by daveverwer** [comment]
I'm not sure what's going wrong with [Nifty-Markdown-Formatter](https://swiftpackageindex.com/NiftyTreeStudios/Nifty-Markdown-Formatter). There's nothing obvious that I can see, and their builds seem to be successful.

Investigating further, this is the build log:

https://gitlab.com/finestructure/swiftpackageindex-builder/-/jobs/2644466709

Line 286 shows the problem:

```
xcodebuild: error: The workspace named "spi-builder-workspace" does not contain a scheme named "NiftyMarkdownFormatter". The "-list" option can be used to find the names of the schemes in the workspace.
[287](https://gitlab.com/finestructure/swiftpackageindex-builder/-/jobs/2644466709#L287)
```

However, their target in the `.spi.yml` does seem to be correct from a quick look at their `Package.swift` file.

**Comment by daveverwer** [comment]
> However, their target in the `.spi.yml` does seem to be correct from a quick look at their `Package.swift` file.

Looks like their scheme is "Nifty Markdown Formatter" not "NiftyMarkdownFormatter":

```
Information about workspace "Nifty-Markdown-Formatter":
    Schemes:
        Nifty Markdown Formatter
```

**Comment by daveverwer** [comment]
I just found out that @finestructure [already made a PR](https://github.com/NiftyTreeStudios/Nifty-Markdown-Formatter/pull/19)! 

Converting this to a discussion where we can keep track of any other documentation failures.

**Comment by finestructure** [comment]
Here are my notes from investigating this issue this week - just hadn't gotten around to sharing them yet, sorry:

## Failing docs
* [x] DesignReviewer
	- was missing iOS platform, submitted PR
* [x] GatheredKit
	- `Error: No documentation target specified`
	- https://gitlab.com/finestructure/swiftpackageindex-builder/-/jobs/2644359247
	- needs to be build on iOS
	- retriggered: https://gitlab.com/finestructure/swiftpackageindex-builder/-/pipelines/573883166
	- still: `Error: No documentation target specified`
	* [x] run locally in builder test
	- platform key was `iOS`, needs to be `ios`
	- PR raised: https://github.com/JosephDuffy/GatheredKit/pull/63
* [x] MuJoCo
	- doesn't build on macos-spm, looks like a genuine build failure
* [x] Nifty Markdown Formatter
	- https://gitlab.com/finestructure/swiftpackageindex-builder/-/jobs/2644347976
	- `Error: No documentation target specified`
	- needs iOS build
	- retriggered: https://gitlab.com/finestructure/swiftpackageindex-builder/-/pipelines/573890776
	- still no S3 doc archives, logs obscured, can't tell what happened
	- run locally in builder test
	- PR opened to fix: https://github.com/NiftyTreeStudios/Nifty-Markdown-Formatter/pull/19
* [x] Script.swift
	* [x] fix doc upload https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/issues/1852
	* [x] re-trigger after fix
* [x] secp256k1
	- same parsing error as Script.swift above
	- PR to fix spi.yml https://github.com/GigaBitcoin/secp256k1.swift/pull/206
	* [x] re-trigger after fix
* [x] StoreFlowable
	- same parsing error as Script.swift above
	* [x] re-trigger after fix
* [x] StreamChatVapor
	- `error: no target named 'stream-chat-vapor-swift'`
	- Opened PR: https://github.com/GetStream/stream-chat-vapor-swift/pull/1
